packages/api/src/context.ts

```diff
@@ -74,6 +77,41 @@ export function createTRPCContext(
         apiClient,
       });
 
+      if (posthog) {
+        const cookies = cookie.parse(opts.req.headers.cookie ?? "");
+
+        let distinctId = cookies["distinctId"];
+
+        if (!distinctId) {
+          distinctId = crypto.randomUUID();
+
+          opts.res.setHeader(
+            "Set-Cookie",
+            cookie.serialize("distinctId", distinctId, {
+              maxAge: 60 * 60 * 24 * 365,
+            })
+          );
+        }
+
+        let ip =
+          opts.req.headers["x-forwarded-for"] ?? opts.req.socket.remoteAddress;
+
+        if (Array.isArray(ip)) {
+          ip = ip[0];
+        }
```

Comment:
let's move this code to its own function

Reply:
Done [f2c9caa](https://github.com/Blobscan/blobscan/pull/578/commits/f2c9caae81817dcae3caf35b96708a2fe8837d5d)